lib/params/FlowParams.cpp

```diff
@@ -382,3 +382,37 @@ int FlowParams::GetSeedInjInterval() const
 }
 
 void FlowParams::SetSeedInjInterval(int val) { SetValueLong(_seedInjInterval, "What's the interval of injecting seeds into an unsteady flow advection", val); }
+
+double FlowParams::_getRakeCenter(int dim)
+{
+    Box *rakeBox = GetRakeBox();
+    if (rakeBox == nullptr) return 0.;
+    std::vector<double> minExt, maxExt;
+    rakeBox->GetExtents(minExt, maxExt);
+    return (maxExt[dim] + minExt[dim]) / 2.;
+}
+
+void FlowParams::_setRakeCenter(int dim, double center)
+{
+    Box *rakeBox = GetRakeBox();
+    if (rakeBox == nullptr) return;
+
+    std::vector<double> minExt, maxExt;
+    rakeBox->GetExtents(minExt, maxExt);
```

Comment:
It's encouraged to use the API with arrays.

Reply:
Fixed